microservices/object_detection/detector.py

```diff
@@ -1,42 +1,39 @@
 from config import PATH_TO_MODEL, THRESHOLD
 import detector_base
+from config import log
 import detect
 from annotator import Annotator
 import bottle
-from bottle import route, run,  get, post, request 
+from bottle import route, run,  get, post, request, response
 import io
 from PIL import Image
 from PIL import ImageDraw
 
 @route('/')
 def greetings():
-    return "Hi, I am the AI4Coral RESTful API\n"
-
+  return "Hi, I am the AI4Coral RESTful API\n"
 
 @post('/detect/<filename>')
 def do_detections(filename):
-    try:
-        try:    
-            interpreter = detector_base.make_interpreter(PATH_TO_MODEL)
-            interpreter.allocate_tensors()
+  try:
+    interpreter = detector_base.make_interpreter(PATH_TO_MODEL)
+    interpreter.allocate_tensors()
 
-            image = Image.open(request.body)
-            scale = detect.set_input(interpreter, image.size,
+    image = Image.open(request.body)
+    scale = detect.set_input(interpreter, image.size,
                            lambda size: image.resize(size, Image.ANTIALIAS))
-            
-            interpreter.invoke()
-            objs = detect.get_output(interpreter, THRESHOLD, scale)
-            
-            ann = Annotator()
-            annotations = ann.get_annotations(objs=objs, filename=filename)
-
-            return annotations
-        except:
-            raise ValueError
-    except ValueError:
-        response.status = 400
-        return
+    interpreter.invoke()
+    objs = detect.get_output(interpreter, THRESHOLD, scale)
 
+    ann = Annotator()
+    annotations = ann.get_annotations(objs=objs, filename=filename)
+    log.info("detections made succefully")
+    return annotations
+  except Exception as e:
+    response.status = 400
+    log.exception("could not make detections")
```

Comment:
We should probably log some information from variable `e` to get more insights. Do we know the type of possible exceptions we can get?

Reply:
the possible types of exceptions will depend on the inputs to the api, and some other random events \
that's why I used  ``` log.exception ```, it will log the message and the exception, is not that cool?

Reply:
What I mean is, if there is an exception, the only thing we will know is "could not make detections". But we have more information on that in the `e` variable: eg what type of exception, the trace tree, maybe some more specific error. Therefore, I'm saying we should take that information and save it, so we can understand better where the exception is coming from (from `Annotator`, `Image`, etc, ?)

Reply:
I already said that ``` log.exception ```, will do the all work, both the message and the exception will be logged ...

Reply:
Ahhhhhh! Now I got it, that's awesome :o I've just checked logging's documentation, I was not aware of that, very cool!

Reply:
You indeed said it before but I guess my bias didn't allow me to see the fact :D